exercises/high-scores/example.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Debug)]
+pub struct HighScores {
+    scores: Vec<u32>,
+}
+
+impl HighScores {
+    pub fn new(scores: &[u32]) -> Self {
+        Self {
+            scores: scores.to_vec(),
```

Comment:
Instead of converting the scores input to an owned vector, this exercise could serve as a simple introduction to lifetimes by retaining a reference to the input slice. I think that direction would improve the quality of this exercise.

Necessary changes to implement that:

- update this example
- add a `.meta/hints.md` file recommending that students retain a reference to the input, and regenerate the readme
- update the topics in `config.json` to include `lifetimes`

Reply:
I did it with lifetimes first, and then reverted. On the second look, it's probably not that difficult with this exercise. I'll update to using references.